src/neo-vm/ExecutionEngine.cs

```diff
@@ -549,61 +551,86 @@ private bool ExecuteInstruction()
                     }
 
                 // Splice
-                case OpCode.CAT:
+                case OpCode.NEWBUFFER:
                     {
-                        if (!TryPop(out ReadOnlyMemory<byte> x2)) return false;
-                        if (!TryPop(out ReadOnlyMemory<byte> x1)) return false;
-                        ByteArray result;
-                        if (x1.IsEmpty)
-                        {
-                            result = x2;
-                        }
-                        else if (x2.IsEmpty)
-                        {
-                            result = x1;
-                        }
-                        else
+                        if (!TryPop(out PrimitiveType x)) return false;
+                        Buffer buffer;
+                        switch (x)
                         {
-                            int length = x1.Length + x2.Length;
-                            if (!CheckMaxItemSize(length)) return false;
-                            byte[] dstBuffer = new byte[length];
-                            x1.CopyTo(dstBuffer);
-                            x2.CopyTo(dstBuffer.AsMemory(x1.Length));
-                            result = dstBuffer;
+                            case Integer i:
+                                int n = i.ToInt32();
+                                if (n < 0 || n > MaxItemSize) return false;
+                                buffer = new Buffer(n);
+                                break;
+                            case ByteArray array:
+                                buffer = new Buffer(array.Span);
+                                break;
+                            default:
+                                return false;
                         }
+                        Push(buffer);
+                        break;
+                    }
+                case OpCode.MEMCPY:
+                    {
+                        if (!TryPop(out int n)) return false;
+                        if (n < 0) return false;
+                        if (!TryPop(out int si)) return false;
```

Comment:
What 'si' means?  We could rename it to make it more self-explanatory.

Reply:
src_index